filebeat/input/filestream/input_integration_test.go

```diff
@@ -0,0 +1,144 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build integration
+
+package filestream
+
+import (
+	"context"
+	"runtime"
+	"sync"
+	"testing"
+
+	loginp "github.com/elastic/beats/v7/filebeat/input/filestream/internal/input-logfile"
+	input "github.com/elastic/beats/v7/filebeat/input/v2"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/logp"
+)
+
+// test_close_renamed from test_harvester.py
+func TestFilestreamCloseRenamed(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("renaming files while Filebeat is running is not supported on Windows")
+	}
+
+	env := newInputTestingEnvironment(t)
+
+	testlogName := "test.log"
+	inp, err := (&loginp.InputManager{
+		Logger:     logp.L(),
+		StateStore: env.stateStore,
+		Type:       pluginName,
+		Configure:  configure,
+	}).Create(common.MustNewConfigFrom(
+		map[string]interface{}{
+			"paths":                                []string{env.abspath(testlogName) + "*"},
+			"prospector.scanner.check_interval":    "1ms",
+			"close.on_state_change.check_interval": "1ms",
+			"close.on_state_change.renamed":        "true",
+		},
+	))
```

Comment:
nit: either use `configure` nor `Plugin`. For example:

```
inp, err := Plugin(logp.L(), env.stateStore).Manager.Create(...)
```

this will give us the same "setup" as in using the code without having to duplicate how the InputManager is created. Already having an `inputTestingEnvironment` we might make it a method of said environment (making sure we have only one input manager instance):

```
func (e *inputTestEnvironment) CreateInput(config map[string]interface{}) (v2.Input, error) {
  manager := e.getManager()
  return manager.Create(config)
}

func (e *inputTestEnvironment) MustCreateInput(config ...) v2.Input {
  inp, err := e.CreateInput(config)
  require.NoError(t, err)
}

func (e *inputTestEnvironment) getManager() v2.InputManager {
  e.managerInitOnce.Do(func() {
    e.plugin = Plugin(e.logger, e.stateStore)
  })
  return e.Manager
}
```